src/Task/ConvertTranslatableTask.php

```diff
@@ -0,0 +1,268 @@
+use TractorCow\Fluent\Task\ConvertTranslatableTask\Exception;
+
+/**
+ * Provides migration from the Translatable module in a SilverStripe 3 website to the Fluent format for SilverStripe 4.
+ * This task assumes that you have upgraded your website to run on SilverStripe 4 already, and you want to migrate the
+ * existing data from your project into a format that is compatible with Fluent.
+ *
+ * Don't forget to:
+ *
+ * 1. Back up your DB
+ * 2. dev/build
+ * 3. Log into the CMS and set up the locales you want to use
+ * 4. Back up your DB again
+ * 5. Log into the CMS and check everything
+ */
+class ConvertTranslatableTask extends BuildTask
+{
+    protected $title = "Convert Translatable > Fluent Task";
+
+    protected $description = "Migrates site DB from SS3 Translatable DB format to SS4 Fluent.";
+
+    private static $segment = 'ConvertTranslatableTask';
+
+    /**
+     * Fields that are localised per class (key = class name)
+     *
+     * @var array[]
+     */
+    protected $localisedFields = [];
+
+    /**
+     * Checks that fluent is configured correctly
+     *
+     * @throws ConvertTranslatableTask\Exception
+     */
+    protected function checkInstalled()
+    {
+        // Assert that fluent is configured
+        $locales = Locale::getLocales();
+        if (empty($locales)) {
+            throw new Exception("Please configure Fluent locales (in the CMS) prior to migrating from translatable");
+        }
+
+        $defaultLocale = Locale::getDefault();
+        if (empty($defaultLocale)) {
+            throw new Exception(
+                "Please configure a Fluent default locale (in the CMS) prior to migrating from translatable"
+            );
+        }
+    }
+
+    /**
+     * Do something inside a DB transaction
+     *
+     * @param callable $callback
+     * @throws \Exception
+     */
+    protected function withTransaction($callback)
+    {
+        try {
+            Debug::message('Beginning transaction', false);
+            DB::get_conn()->transactionStart();
+            $callback($this);
+            Debug::message('Comitting transaction', false);
+            DB::get_conn()->transactionEnd();
+        } catch (\Exception $ex) {
+            Debug::message('Rolling back transaction', false);
+            DB::get_conn()->transactionRollback();
+            throw $ex;
+        }
+    }
+
+    protected $translatedFields = [];
+
+    /**
+     * Get all database fields to translate
+     *
+     * @param string $class Class name
+     * @return array List of translated fields
+     */
+    public function getTranslatedFields($class)
+    {
+        if (isset($this->localisedFields[$class])) {
+            return $this->localisedFields[$class];
+        }
+        $fields = [];
+        $hierarchy = ClassInfo::ancestry($class);
+        foreach ($hierarchy as $class) {
+            // Skip classes without tables
+            if (!DataObject::getSchema()->classHasTable($class)) {
```

Comment:
You can filter this with `ClassInfo::ancestry($class, true)` instead of manually.

Reply:
We actually don't need this method any more. The Fluent ORM handles it internally